.github/actions/verify-token/src/utils.ts

```diff
@@ -58,14 +60,30 @@ export function getTriggerRef(rawTokenObj: rawTokenInterface): string {
   return rawTokenObj.github.ref;
 }
 
+function validateSha1(sha1: string): void {
+  // Length of sha1 in hex format must be 40.
+  if (sha1.length !== 40) {
+    throw new Error(`invalid sha1 length: ${sha1.length}`);
+  }
+
+  // Check if the string only contains hexadecimal characters.
+  for (const c of sha1) {
+    if (!/[a-fA-F0-9]/.test(c)) {
```

Comment:
you can add the length check to the regex
```suggestion
    if (!/[a-fA-F0-9]{40}/.test(c)) {
```